### ISSUE TITLE
Segmentation fault in creation of SOAP descriptor

### ISSUE THREAD

**Issue by OHANAN1** [SINGROUP/dscribe#139]
Dear Dscribe-Maintainers,

I have encountered an issue when creating SOAP descriptors from crystal structures.
When a ase.Atoms object contains a NaN value in one of its position vectors a `Segmentation fault (core dumped)`  error occurs.
While it makes sense that this ase.Atoms object should not produce a Descriptor,  a python error should be raised to allow
for correct handling of it or to better debug the script. 

Code to reproduce the segmentation fault:
```python
from dscribe.descriptors import SOAP
import numpy as np
import ase

crystal = ase.Atoms(
    symbols=["Si", "Sn", "Ge"],
    positions=[[0, 0, 0], [5, 0, 0], [5, 5, np.nan]],
    cell=[5, 5, 5],
    pbc=True,
)

soap_object = SOAP(
    species=["Si", "Sn", "Ge"],
    r_cut=15,
    n_max=8,
    l_max=8,
    periodic=True,
    sparse=False,
    average="inner",
)

features = soap_object.create(crystal)
print(features)
```
**Expected behaviour**:  

An python error should be raised that states that a SOAP can not be created for a ase.Atoms object with incorrect positions.

**Actual behaviour**:

Script gets terminated and segmentation fault is raised, which makes handeling and backtracing hard.

**Environment**:
- python version: 3.10.12
- dscribe version: 2.1.0
- ase version: 3.22.1
- numpy version: 1.26.4

Thank you for your work on dscribe!

**Comment by lauri-codes** [comment]
Hi @OHANAN1.

Thanks for the report. This is a bit tricky. I know that in principle we could just run something like `np.isfinite` on the input coordinates and raise exceptions accordingly. But this raises some concerns:

- Is there a performance penalty, since positions can be very large (imagine a system with million atoms).
- Why is this not done already by ASE?
- What else should we check: cells, species, all SOAP parameters?

Because of these concerns, I would rather not implement this check at the moment. If this becomes a commonly occurring problem for multiple users, I could imagine adding this check.

**Comment by OHANAN1** [comment]
Hi @lauri-codes,

thank you for your response.
while I completely accept your decisions as a maintainer,  I  would like to share some thoughts.

Regarding your first concerns, I wrote a test script for the chase of 1.000.000 position vectors. 
``` python
import time
import ase
import numpy as np

size = 1000000

positions = np.random.uniform(0, 100, (size, 3))
numbers = np.random.randint(0, 20, size)

example_crystal = ase.Atoms(
    numbers=numbers,
    positions=positions,
    cell=[size, size, np.nan],
    pbc=True,
)


start = time.time()
print("isnan:", np.isnan(example_crystal.get_positions()).any())
print(time.time() - start, "seconds")


positions[0] = [np.nan, np.nan, np.nan]
example_crystal.set_positions(positions)

start = time.time()
print("isnan:", np.isnan(example_crystal.get_positions()).any())
print(time.time() - start, "seconds")
```
Output:
```
isnan: False
0.00804591178894043 seconds
isnan: True
0.008104562759399414 seconds
```

In the chase of one million position vectors the use of np.nan is still pretty efficient, especially compared to the calculation time needed to calculate the SOAP Descriptor of this kind of crystal.

Regarding your question, why ASE does not take care of this, I think that this it is an intentional feature.
For example the slap object takes a empty ase.Atoms object as an input. 
Or one could use this feature to initialize a partly empty Atoms object that will be filled in a later step of the algorithm.

I definitely understand your last concern. I the version of ASE I gave in my first message, the ase.Atoms object can not be initialized with a np.nan entry in the atomic number parameter. While it is possible to initialize a cell with np.nan entries, the cell has at maximum 9 parameters. 

I completely understand if you still decide to not add this check for now, but I think a note about this, somewhere in the docs would be nice.

Thank you very much!

**Comment by lauri-codes** [comment]
Hi @OHANAN1,

Thank you for the additional details. I think your answer does clear my initial doubts about this check. In the next release, I can try to implement this as a generic check that all descriptors do before accepting a system.

**Comment by lauri-codes** [comment]
This check is now included in v2.1.1.
